Makefile

```diff
@@ -1,4 +1,7 @@
 ifneq ($O,)
+ifneq ($(shell echo $(O) | grep -- "/\.$$"),)
+$(error Cannot support output directory 'O' path ending with terminal subpath info "./")
+endif
```

Comment:
ok, i must fix this. 
the thing is: rmdir does not like pathes as some/path/. or /some/path/./.
from your proposal below, we can even discard the restriction above.